Source/Core/Core/IOS/Network/IP/Top.cpp

```diff
@@ -621,55 +619,24 @@ IPCCommandResult NetIPTop::HandlePollRequest(const IOCtlRequest& request)
   for (int i = 0; i < nfds; ++i)
   {
     s32 wii_fd = Memory::Read_U32(request.buffer_out + 0xc * i);
-    ufds[i].fd = WiiSockMan::GetInstance().GetHostSocket(wii_fd);          // fd
+    ufds[i].fd = sm.GetHostSocket(wii_fd);                                 // fd
     int events = Memory::Read_U32(request.buffer_out + 0xc * i + 4);       // events
     ufds[i].revents = Memory::Read_U32(request.buffer_out + 0xc * i + 8);  // revents
 
     // Translate Wii to native events
-    int unhandled_events = events;
-    ufds[i].events = 0;
-    for (auto& map : mapping)
-    {
-      if (events & map.wii)
-        ufds[i].events |= map.native;
-      unhandled_events &= ~map.wii;
-    }
+    ufds[i].events = WiiSockMan::ConvertEvents(events, WiiSockMan::ConvertDirection::WiiToNative);
     DEBUG_LOG(IOS_NET,
               "IOCTL_SO_POLL(%d) "
-              "Sock: %08x, Unknown: %08x, Events: %08x, "
+              "Sock: %08x, Events: %08x, "
               "NativeEvents: %08x",
-              i, wii_fd, unknown, events, ufds[i].events);
+              i, wii_fd, events, ufds[i].events);
 
     // Do not pass return-only events to the native poll
     ufds[i].events &= ~(POLLERR | POLLHUP | POLLNVAL | UNSUPPORTED_WSAPOLL);
-
-    if (unhandled_events)
-      ERROR_LOG(IOS_NET, "SO_POLL: unhandled Wii event types: %04x", unhandled_events);
   }
 
-  int ret = poll(ufds.data(), nfds, timeout);
-  ret = WiiSockMan::GetNetErrorCode(ret, "SO_POLL", false);
-
-  for (int i = 0; i < nfds; ++i)
-  {
-    // Translate native to Wii events
-    int revents = 0;
-    for (auto& map : mapping)
-    {
-      if (ufds[i].revents & map.native)
-        revents |= map.wii;
-    }
-
-    // No need to change fd or events as they are input only.
-    // Memory::Write_U32(ufds[i].fd, request.buffer_out + 0xc*i); //fd
-    // Memory::Write_U32(events, request.buffer_out + 0xc*i + 4); //events
-    Memory::Write_U32(revents, request.buffer_out + 0xc * i + 8);  // revents
-
-    DEBUG_LOG(IOS_NET, "IOCTL_SO_POLL socket %d wevents %08X events %08X revents %08X", i, revents,
-              ufds[i].events, ufds[i].revents);
-  }
-
-  return GetDefaultReply(ret);
+  sm.AddPollCommand({request.address, request.buffer_out, std::move(ufds), timeout});
```

Comment:
Maybe you should leave a comment to explain the reason behind this as SO_POLL isn't async and returns a reply. Something along those lines:
```C++
// Prevent to block the emulation on blocking poll
```